### PR TITLE
Fix two issues with writing private key with passphrase

### PR DESCRIPTION
**Pull request by cjh1**
See #568

### CODE REVIEW

File: paramiko/pkey.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def _write_private_key(self, f, key, format, password=None):
         if password is None:
             encryption = serialization.NoEncryption()
         else:
-            encryption = serialization.BestEncryption(password)
+            encryption = serialization.BestAvailableEncryption(password.encode())
```
Comment (alexpilotti) [inline paramiko/pkey.py]: The method does not specify if the "password" parameter is bytes or string. If it's not documented elsewhere, wouldn't it be better to expect bytes in input? This affects python3.

**Comment by alexpilotti** [comment]
This fixes issue #741

File: paramiko/pkey.py (hunk 1 above)
Comment (cjh1) [inline paramiko/pkey.py]: All the public methods in this class that take a password are expecting strings, so I think this password parameter should also be a string. Should I add a doc string to indicate this?

**Comment by bitprophet** [comment]
Rolling into #741.

**Comment by cjh1** [comment]
So have these fixes been merged?

**Comment by ploxiln** [comment]
yes, they'll be in the next 2.y.z releases

**Comment by bitprophet** [comment]
Been laid low today but yea I'm planning to release all this stuff as soon as I can. May toss in some low hanging fruit from the bugfix milestone too, but there's already so much ready to go out I don't absolutely have to.

**Comment by cjh1** [comment]
Excellent! Many Thanks

**Pull request closed without merging**